src/qmapshack/gis/rte/router/CRouterRoutino.cpp

```diff
@@ -248,18 +245,67 @@ void CRouterRoutino::buildDatabaseList()
                 continue;
             }
 
+            // qDebug() << "buildDatabase Prefix" << prefix;
+
 #ifdef Q_OS_WIN
             Routino_Database * data = Routino_LoadDatabase(dir.absolutePath().toLocal8Bit(), prefix.toLocal8Bit());
 #else
             Routino_Database * data = Routino_LoadDatabase(dir.absolutePath().toUtf8(), prefix.toUtf8());
 #endif
-            if(data)
+            qDebug() << "Loaded Routino DB" << dir.absolutePath().toUtf8().data() << "  " << prefix.toUtf8().data();
+
+            if(data == nullptr)
             {
-                comboDatabase->addItem(prefix.replace("_", " "), quint64(data));
+                QMessageBox::critical(this, "Routino ...", xlateRoutinoError(Routino_errno), QMessageBox::Abort);
+                continue;
+            }
+            /* determine the profile to use for each database*/
+            QVariantMap dmap;
+            dmap["db"] = QVariant ((qulonglong)data);
+
+            /* check possible profiles.xml locations and use the first available */
+            int pError = 0;
+            dmap["profilesPath"] = "";
+            QStringList profilesPaths = {
+                dir.filePath(prefix+"-profiles.xml"),
+                dir.filePath("profiles.xml"),
+                setup->routinoPath("profiles.xml").toUtf8()
+            };
+            for(const QString& profilePath : profilesPaths)
+            {
+                QFileInfo pinfo(profilePath);
+                if( pinfo.isReadable())
+                {
+                    dmap["profilesPath"] = pinfo.filePath();
+                    break;
+                }
+            }
+            if( dmap["profilesPath"].toString().isEmpty() )
+            {
+                QMessageBox::critical(this, "Routino...",tr("Could not find a profiles XML file in expected folders. Routino Routing will not function"), QMessageBox::Ok);
+                pError = 1;
+            }
+            else
+            {
+                /* check if profile will load - will abort if not good */
+                pError = loadProfiles(dmap["profilesPath"].toString());
+            }
+
+            qDebug() << "Profile ... Using \n" << dmap["prefixpath"].toString();
```

Comment:
I guess this should be `profilesPath` instead of  `prefixpath`

Reply:
Yes thanks.